addons/hide-backpack/userscript.js

```diff
@@ -0,0 +1,86 @@
+export default async function ({ addon }) {
+  let originalBackpack = await getBackpackElement();
+  changeBackpackVisibility();
+
+  // Event listeners that add dynamic enable/disable + setting change
+  addon.settings.addEventListener("change", () => changeBackpackVisibility(addon));
+  addon.self.addEventListener("reenabled", () => changeBackpackVisibility(addon));
+  addon.self.addEventListener("disabled", () => {
+    moveResizeButtons(0);
+    originalBackpack.style.display = "block";
+    window.dispatchEvent(new Event("resize"));
+    document.querySelector(".sa-backpack-button").style.display = "none";
+  });
+
+  while (true) {
+    originalBackpack = await getBackpackElement();
+    if (!addon.self.disabled) changeBackpackVisibility();
+  }
+
+  async function getBackpackElement() {
```

Comment:
Does this need to be its own function? I don't think the call at the top is necessary.